source/core_mqtt.c

```diff
@@ -1289,20 +1290,44 @@ static MQTTStatus_t sendPublishAcks( MQTTContext_t * pContext,
                                     packetTypeByte,
                                     packetId );
 
+        if( status == MQTTSuccess )
+        {    
+            MQTT_PRE_STATE_UPDATE_HOOK( pContext );
+
+            stateUpdateHookExecuted = true;
+
+            if( pContext->connectStatus != MQTTConnected)
+            {
+                status = MQTTDisconnected;
+            }
+        }
+
         if( status == MQTTSuccess )
         {
-            MQTT_PRE_SEND_HOOK( pContext );
+            // MQTT_PRE_SEND_HOOK( pContext );
```

Comment:
The Pre and Post Send hook Macros are not required now, as the sending logic will be within the pre and post state update hook itself. (because we cannot allow other threads to change the connection state of the application until a send operation is complete). 

And yes the commented out code will be removed, noted.